src/components/split-axis-events.js

```diff
@@ -1,30 +1,376 @@
+      this.actuator = trackedControls.controller.hapticActuators[0];
+    } else {
+      setTimeout(this.tryGetActuator, 1000);
+    }
+  },
+
+  play: function() {
+    this.el.addEventListener(`${this.data.hand}_haptic_pulse`, this.pulse);
+  },
+  pause: function() {
+    this.el.removeEventListener(`${this.data.hand}_haptic_pulse`, this.pulse);
+  },
+
+  pulse: function(event) {
+    let { strength, duration, intensity } = event.detail;
+    if (intensity === "low") {
+      strength = 0.07;
+      duration = 12;
+    }
+    if (intensity === "medium") {
+      strength = 0.2;
+      duration = 12;
+    }
+    if (intensity === "high") {
+      strength = 1;
+      duration = 12;
+    }
+    if (intensity === "none") {
+      return;
+    }
+
+    this.actuator.pulse(strength, duration);
+  }
+});
+
+AFRAME.registerComponent("vive-controls-extended", {
+  schema: {
+    hand: { default: "left" },
+    dpad: { default: false },
+    dpad_livezone: { default: 0.3 },
+    dpad_deadzone: { default: 0.8 },
+    dpad_directions: { default: 4 },
+    dpad_turbo: { default: true },
+    dpad_pressed_turbo: { default: false },
+    center_zone: { default: 0.3 },
+    dpad_haptic_intensity: { default: "none" },
+    dpad_pressed_haptic_intensity: { default: "none" }
+  },
+
   init: function() {
-    this.pressed = false;
+    this.dpadCanFire = true;
+    this.dpadPressedCanFire = true;
+    this.trackpadPressed = false;
     this.onAxisMove = this.onAxisMove.bind(this);
+    this.lastSeenAxes = [0, 0];
     this.onButtonChanged = this.onButtonChanged.bind(this);
   },
 
   play: function() {
     this.el.addEventListener("axismove", this.onAxisMove);
-    this.el.addEventListener("buttonchanged", this.onButtonChanged);
+    this.el.addEventListener("trackpadchanged", this.onButtonChanged);
   },
 
   pause: function() {
     this.el.removeEventListener("axismove", this.onAxisMove);
-    this.el.removeEventListener("buttonchanged", this.onButtonChanged);
+    this.el.removeEventListener("trackpadchanged", this.onButtonChanged);
   },
 
   onAxisMove: function(event) {
-    var name = "touchpad" + (this.pressed ? "pressed" : "") + "axismove";
-    this.el.emit(name + "x", event.detail.axis[0]);
-    this.el.emit(name + "y", event.detail.axis[1]);
+    var x = event.detail.axis[0];
+    var y = event.detail.axis[1];
+    this.lastSeenAxes = [x, y];
+    var hand = this.data.hand;
+    const axisMoveEvent = `${hand}_trackpad${this.trackpadPressed
+      ? "_pressed"
+      : ""}_axismove`;
+    this.el.emit(axisMoveEvent, {
+      axis: [x, y]
+    });
+
+    var deadzone = this.data.dpad_deadzone;
+    var turbo = this.data.dpad_turbo;
+    var pressedTurbo = this.data.dpad_pressed_turbo;
+    var livezone = this.data.dpad_livezone;
+    var directions = this.data.dpad_directions;
+    var hapticIntensity = this.data.dpad_haptic_intensity;
+    var pressedHapticIntensity = this.data.dpad_pressed_haptic_intensity;
```

Comment:
Use `let` instead of `var`. Though it looks like you can use `const` for most of these.